### PR TITLE
Start moving to the rustc guide!

### PR DESCRIPTION
**Pull request by mark-i-m**
r? @nikomatsakis 

cc #48478

### CODE REVIEW

File: src/librustc/README.md [hunk 1]
```diff
@@ -1,204 +0,0 @@
-An informal guide to reading and working on the rustc compiler.
```
Comment (nikomatsakis) [inline src/librustc/README.md]: Can you replace this with a README -- or maybe doc-comment on the crate -- that says to see the rustc guide? I know it's in other places, but I want to make it as visible as possible.

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
r=me after that one nit

**Comment by nikomatsakis** [comment]
Oh, one other thing. I think there may be references to those deleted READMEs in doc-comments and things. Could you maybe redirect those to the appropriate URL in the rustc guide?

Examples:

https://github.com/rust-lang/rust/blob/063deba92e44809125a433ca6e6c1ad0993313bf/src/librustc/lib.rs#L14

https://github.com/rust-lang/rust/blob/063deba92e44809125a433ca6e6c1ad0993313bf/src/librustc/lib.rs#L32

**Comment by nikomatsakis** [comment]
I found those by doing `rg README.md | grep '.rs'`

**Comment by mark-i-m** [comment]
@nikomatsakis I think it should be fixed now... I also updated the list in rust-lang-nursery/rustc-guide#2... I think a lot of those will be closed by your latest PR, but it can't hurt to list them...

**Pull request merged**